File: dotnet/GitHub/Repos/Item/Item/Pulls/Item/Merge/MergePutRequestBody.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System;
namespace GitHubTodoDemo.GitHub.Repos.Item.Item.Pulls.Item.Merge {
    public class MergePutRequestBody : IAdditionalDataHolder, IParsable {
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; set; }
        /// <summary>Extra detail to append to automatic commit message.</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? CommitMessage { get; set; }
#nullable restore
#else
        public string CommitMessage { get; set; }
#endif
        /// <summary>Title for the automatic commit message.</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? CommitTitle { get; set; }
#nullable restore
#else
        public string CommitTitle { get; set; }
#endif
        /// <summary>The merge method to use.</summary>
        public MergePutRequestBody_merge_method? MergeMethod { get; set; }
        /// <summary>SHA that pull request head must match to allow merge.</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Sha { get; set; }
#nullable restore
#else
        public string Sha { get; set; }
#endif
        /// <summary>
        /// Instantiates a new mergePutRequestBody and sets the default values.
        /// </summary>
        public MergePutRequestBody() {
            AdditionalData = new Dictionary<string, object>();
        }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static MergePutRequestBody CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new MergePutRequestBody();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"commit_message", n => { CommitMessage = n.GetStringValue(); } },
                {"commit_title", n => { CommitTitle = n.GetStringValue(); } },
                {"merge_method", n => { MergeMethod = n.GetEnumValue<MergePutRequestBody_merge_method>(); } },
                {"sha", n => { Sha = n.GetStringValue(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteStringValue("commit_message", CommitMessage);
            writer.WriteStringValue("commit_title", CommitTitle);
            writer.WriteEnumValue<MergePutRequestBody_merge_method>("merge_method", MergeMethod);
            writer.WriteStringValue("sha", Sha);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}

File: php/GitHub/Repos/Item/Item/Pulls/Item/Merge/MergePutRequestBody.php
<?php

namespace Baywet\Githubtododemo\Github\Repos\Item\Item\Pulls\Item\Merge;

use Microsoft\Kiota\Abstractions\Serialization\AdditionalDataHolder;
use Microsoft\Kiota\Abstractions\Serialization\Parsable;
use Microsoft\Kiota\Abstractions\Serialization\ParseNode;
use Microsoft\Kiota\Abstractions\Serialization\SerializationWriter;

class MergePutRequestBody implements AdditionalDataHolder, Parsable 
{
    /**
     * @var array<string, mixed>|null $additionalData Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    */
    private ?array $additionalData = null;
    
    /**
     * @var string|null $commit_message Extra detail to append to automatic commit message.
    */
    private ?string $commit_message = null;
    
    /**
     * @var string|null $commit_title Title for the automatic commit message.
    */
    private ?string $commit_title = null;
    
    /**
     * @var MergePutRequestBody_merge_method|null $merge_method The merge method to use.
    */
    private ?MergePutRequestBody_merge_method $merge_method = null;
    
    /**
     * @var string|null $sha SHA that pull request head must match to allow merge.
    */
    private ?string $sha = null;
    
    /**
     * Instantiates a new mergePutRequestBody and sets the default values.
    */
    public function __construct() {
        $this->setAdditionalData([]);
    }

    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param ParseNode $parseNode The parse node to use to read the discriminator value and create the object
     * @return MergePutRequestBody
    */
    public static function createFromDiscriminatorValue(ParseNode $parseNode): MergePutRequestBody {
        return new MergePutRequestBody();
    }

    /**
     * Gets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @return array<string, mixed>|null
    */
    public function getAdditionalData(): ?array {
        return $this->additionalData;
    }

    /**
     * Gets the commit_message property value. Extra detail to append to automatic commit message.
     * @return string|null
    */
    public function getCommitMessage(): ?string {
        return $this->commit_message;
    }

    /**
     * Gets the commit_title property value. Title for the automatic commit message.
     * @return string|null
    */
    public function getCommitTitle(): ?string {
        return $this->commit_title;
    }

    /**
     * The deserialization information for the current model
     * @return array<string, callable(ParseNode): void>
    */
    public function getFieldDeserializers(): array {
        $o = $this;
        return  [
            'commit_message' => fn(ParseNode $n) => $o->setCommitMessage($n->getStringValue()),
            'commit_title' => fn(ParseNode $n) => $o->setCommitTitle($n->getStringValue()),
            'merge_method' => fn(ParseNode $n) => $o->setMergeMethod($n->getEnumValue(MergePutRequestBody_merge_method::class)),
            'sha' => fn(ParseNode $n) => $o->setSha($n->getStringValue()),
        ];
    }

    /**
     * Gets the merge_method property value. The merge method to use.
     * @return MergePutRequestBody_merge_method|null
    */
    public function getMergeMethod(): ?MergePutRequestBody_merge_method {
        return $this->merge_method;
    }

    /**
     * Gets the sha property value. SHA that pull request head must match to allow merge.
     * @return string|null
    */
    public function getSha(): ?string {
        return $this->sha;
    }

    /**
     * Serializes information the current object
     * @param SerializationWriter $writer Serialization writer to use to serialize this model
    */
    public function serialize(SerializationWriter $writer): void {
        $writer->writeStringValue('commit_message', $this->getCommitMessage());
        $writer->writeStringValue('commit_title', $this->getCommitTitle());
        $writer->writeEnumValue('merge_method', $this->getMergeMethod());
        $writer->writeStringValue('sha', $this->getSha());
        $writer->writeAdditionalData($this->getAdditionalData());
    }

    /**
     * Sets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param array<string,mixed> $value Value to set for the AdditionalData property.
    */
    public function setAdditionalData(?array $value): void {
        $this->additionalData = $value;
    }

    /**
     * Sets the commit_message property value. Extra detail to append to automatic commit message.
     * @param string|null $value Value to set for the commit_message property.
    */
    public function setCommitMessage(?string $value): void {
        $this->commit_message = $value;
    }

    /**
     * Sets the commit_title property value. Title for the automatic commit message.
     * @param string|null $value Value to set for the commit_title property.
    */
    public function setCommitTitle(?string $value): void {
        $this->commit_title = $value;
    }

    /**
     * Sets the merge_method property value. The merge method to use.
     * @param MergePutRequestBody_merge_method|null $value Value to set for the merge_method property.
    */
    public function setMergeMethod(?MergePutRequestBody_merge_method $value): void {
        $this->merge_method = $value;
    }

    /**
     * Sets the sha property value. SHA that pull request head must match to allow merge.
     * @param string|null $value Value to set for the sha property.
    */
    public function setSha(?string $value): void {
        $this->sha = $value;
    }

}
